lib/reports/subscriber_lists_report_row.rb

```diff
@@ -0,0 +1,81 @@
+class Reports::SubscriberListsReportRow
+  def initialize(date, headers, list)
+    @date = date
+    @headers = headers
+    @list = list
+  end
+
+  def call
+    headers.map { |header| send(header) }
```

Comment:
What's the advantage of doing this versus:

```ruby

	headers.map { |header| &:header }
```

(...apart from being confusingly named)?

Reply:
I'm not sure I understand your question...it's calling send on each member of the array `headers` - ie so headers might be `[:title, :slug]` - line 9 simply iterates over the array and calls the relevant function by passing the member of the array to the `send` function, which calls the relevant function (ie for the element `:title` the `title` method of the class will get called. AFAIK calling `&:header` will just try to call the header function for every member of the array - or have I misunderstood something?

Reply:
Ahhhh sorry I didn't grok this properly. Thanks for the explainer. And validate headers stops the report entirely?

Reply:
Yup. I figured if someone was entered the wrong name by accident it's better to tell them than not include that header (which might lead to a lot of head scratching)